internal/clients/team5/forage.go

```diff
@@ -0,0 +1,407 @@
+Born in the middle class
+Roll randomly
+If we succeed we are JB and will risk more to gain more
+If we lose again then we are in the Imperial class and we fish to try to get back to middle class
+*/
+func (c *client) InitialForage() shared.ForageDecision {
+	// Figure out how much the cost of livin per turn is
+	//=============================================================================
+	// if c.getTurn() == 2 { // On turn 2
+	// 	var turncost shared.Resources
+	// 	for _, outcomes := range c.forageHistory { // For each foraging type find the outcome
+	// 		for _, returns := range outcomes { // For each outcome find the returns
+	// 			turncost += returns.input // cumlative sum of the return on investment (should only be 1 return from foraging)
+	// 			c.Logf("[DEBUG] %v - %v + %v", c.resourceHistory[1], c.resourceHistory[2], returns.input)
+	// 		}
+	// 	}
+	// 	costOfTurn1 := c.resourceHistory[1] - c.resourceHistory[2] + turncost
+	// 	c.Logf("[Debug] Cost per turn to live %v = %v - %v + %v", costOfTurn1, c.resourceHistory[1], c.resourceHistory[2], turncost)
+	// }
+	// =============================================================================
+	var forageType shared.ForageType
+	forageContribution := shared.Resources(c.config.MinimumForagePercentage+
+		rand.Float64()*
+			(c.config.NormalForagePercentage-c.config.MinimumForagePercentage)) *
+		c.gameState().ClientInfo.Resources // Random amount between Min and Normal contribution
+	switch c.wealth() {
+	case jeffBezos: // Rich
+		forageContribution = shared.Resources(c.config.NormalForagePercentage+
+			rand.Float64()*
+				(c.config.JBForagePercentage-c.config.NormalForagePercentage)) *
+			c.gameState().ClientInfo.Resources // JB then we have so much might as well gamble Normal % -> JB% of it
+		forageType = shared.DeerForageType
+	case imperialStudent:
+		forageType = shared.FishForageType // Save money by fishing and hope that we can get some returns
+	case dying:
+		c.lastHopeForage() // Invest all our money into fishing to hope we can get some return
+	case middleClass: // Midle class (lets see where the coin takes us)
+		if rand.Float64() < 0.50 { // Coin
+			forageType = shared.DeerForageType
+		} else {
+			forageType = shared.FishForageType
+		}
+	}
+
+	c.Logf("[Initial Forage]:[%v][%v]", forageType, forageContribution)
+
+	return shared.ForageDecision{
+		Type:         forageType,
+		Contribution: forageContribution,
+	}
+}
+
+// bestForagingType indicates the best foraging method best of RoI (Return on Investment Output/Input - 1)
+func (c *client) bestHistoryForaging(forageHistory forageHistory) shared.ForageType {
+	bestForagingMethod := shared.ForageType(-1) // Default is that there is no good method
+	bestReturn := 0.0
+
+	for forageType, outcomes := range forageHistory { // For each foraging type
+		returnOI := 0.0
+		for _, returns := range outcomes {
+			returnOI += float64((returns.output / returns.input) - 1) // Cumlative sum of the return on investment
```

Comment:
do a quick search for `/` to check if you've checked div0?

Reply:
Will do 👍